packages/terra-functional-testing/src/terra-dev-site/terra-functional-testing/guides/Migration.tool.md

```diff
@@ -0,0 +1,519 @@
+
+### Assertion Framework
+
+We've removed the [chai assertion framework](https://www.chaijs.com/) and opted into using the built in [assertion library](https://v6.webdriver.io/docs/assertion.html) that ships with WebDriverIO. This new assertion framework closely resembles the testing environment most of us are used to in Jest, because it is [Jest's expect library](https://jestjs.io/docs/en/expect). This should help align testing between Jest and WebDriverIO.
+
+A few common changes look something like this:
+
+```diff
+- expect(true).to.be.true; // Chai assertion
++ expect(true).toBe(true); // Expect/Jest assertion.
+
+- expect(true).to.be.false; // Chai assertion
++ expect(true).toBe(false); // Expect/Jest assertion.
+
+- expect(1).to.equal(1); // Chai assertion
++ expect(1).toEqual(1); // Expect/Jest assertion.
+```
+
+WebDriverIO provides a collection of [matchers](https://v6.webdriver.io/docs/api/expect.html) that can be used for assertions in your tests.
+
+### Terra Commands
+
+The `Terra.it` commands have been removed. We noticed they encouraged anti-patterns while writing tests and made testing more confusing.
+
+These commands have been removed:
+
+- Terra.it.validatesElement
+- Terra.it.isAccessible
+- Terra.it.matchesScreenshot
+
+These commands can still be used but have had some changes:
+
+- Terra.validates.accessibility
+- Terra.validates.element
+- Terra.validates.screenshot
+
+The validates screenshot and validates element commands now require a screenshot name.
+
+```js
+it('should click and validate the element', () => {
+  $('#element').click();
+
+  Terra.validates.element('screenshot name'); //  A unique screenshot name must be provided or the command will throw an error.
+});
+```
+
+The `viewports` option has been removed from `Terra.validates.screenshot`. The viewport option array created additional viewport permutations within other viewport permutations. To more consistently align viewport testing the option was removed. To specify an array of viewports use the `Terra.describeViewports` helper.
+
+```diff
++ Terra.describeViewports('Terra.validates', ['tiny', 'small'], () => {
+   it('should click and validate the element', () => {
+     $('#element').click();
+
+-     Terra.validates.screenshot('screenshot name', { viewports: ['tiny', 'small'] });
++     Terra.validates.screenshot('screenshot name');
+   });
++ });
+```
+
+### Screenshots
+
+We've forked and taken ownership of the visual regression service to give us better control of making changes when necessary. Over time we'll be working to resolve common errors encountered with the visual regression service. Such as the notorious x,y range error.
+
+For now though, there are a few breaking changes to the screenshot strategy that will require regenerating screenshots.
+
+- Screenshot names are no longer auto-generated using the describe blocks.
+- Screenshots now require a unique name be provided.
```

Comment:
Should we clarify that the name only needs to be unique within the spec file? The same screenshot name can be used as long as it is not used for the same theme, locale, viewport size, and spec file. It a little tricky to describe this concisely.

Reply:
I took a swing at clarifying:

https://github.com/cerner/terra-toolkit/pull/574/commits/fb337feddb2c24866952c8a5f2ac8b7f5320b5b4